### PR TITLE
build: add .PHONY to all phony targets on top-level makefile

### PR DESCRIPTION
**Pull request by hunchback**
some of the top-level Makefile were not set as .PHONY (issues discovered on the devmode environment) , this has been addressed in this PR.

### CODE REVIEW

**Comment by safchain** [comment]
ok to test

**Comment by safchain** [comment]
rerun functional-tests-backend-elasticsearch

File: Makefile [hunk 1]
```diff
@@ -129,18 +130,22 @@ BINDATA_DIRS := \
...
 	gofmt -w -s statics/bindata.go
 
+.PHONY: compile
 compile:
 	$(GOVENDOR) install \
 		-ldflags="-X $(SKYDIVE_GITHUB_VERSION)" \
 		${GOFLAGS} -tags="${BUILDTAGS}" ${VERBOSE_FLAGS} \
 		${SKYDIVE_GITHUB}
 
+.PHONY: skydive
```
Comment (nplanel) [inline Makefile]: phony will not apply here as skydive file will be created by the rule

**Comment by safchain** [comment]
rerun functional-tests-backend-elasticsearch

**Pull request merged**